src/meta/api/src/share_api_impl.rs

```diff
@@ -920,7 +918,45 @@ async fn get_share_database_name(
     }
 }
 
-async fn get_outbound_shared_accounts_by_name(
+async fn get_outbound_share_tenants_by_name(
+    kv_api: &(impl KVApi + ?Sized),
+    share_name: &ShareNameIdent,
+) -> Result<Vec<GetShareGrantTenants>, MetaError> {
+    let res = get_share_or_err(
+        kv_api,
+        share_name,
+        format!("get_share: {}", share_name.clone()),
+    )
+    .await?;
+    let (_share_id_seq, share_id, _share_meta_seq, share_meta) = res;
+
+    let mut accounts = vec![];
+    for account in share_meta.get_accounts() {
+        let share_account_key = ShareAccountNameIdent {
+            account: account.clone(),
+            share_id,
+        };
+
+        let (_seq, meta) = get_share_account_meta_or_err(
```

Comment:
I don't like APIs like `get_share_account_meta_or_err` because we have to construct the error string no matter this operator succeed or not.